### PR TITLE
Add support for the non-free netMHC tool family

### PR DESCRIPTION
**Pull request by lkuchenb**
## PR checklist

- [x] This comment contains a description of changes (with reason)
- [x] `CHANGELOG.md` is updated
- [x] If you've fixed a bug or added code that should be tested, add tests!
- [x] Documentation in `docs` is updated
- [x] If necessary, also make a PR on the [nf-core/epitopeprediction branch on the nf-core/test-datasets repo](https://github.com/nf-core/test-datasets/pull/new/nf-core/epitopeprediction)

## Additions

This patch adds support for the non-free 3rd party tools NetMHC, NetMHCpan, NetMHCII and NetMHCIIpan. Since the tools' licenses do not allow for redistribution of the software, the user has to provide the software tarball which can be obtained from the upstream website.

* To maintain reproducibility, the pipeline validates the checksum of the provided software tarballs
* The patch provides a process that
  * Unpacks the software tarballs
  * Performs necessary modifications to the tools wrapper scripts as required by the installation instructions
  * Downloads the model data as required by installation instructions. Again, checksums are being checked.

## Changes

* `epaa.py`: Make tool name comparison case insensitive. Unfortunately, the method names aren't uniform w.r.t. upper / lower case letters among the netMHC family of tools.
* `main.nf`: Convert `params.tools` to a groovy list before parsing. The previous `.contains()` calls were applied on strings, which worked fine for the previously used tools, but since e.g. `netmhc` is a substring of `netmhcpan` it would clash with the newly added tools

### CODE REVIEW

**Comment by lkuchenb** [comment]
This is ready for review from my end. It's worth noting that the added CI will fail when the PR comes from a fork, i.e. only nf-core team members who can open a branch here can also run the CI successfully. Thus I guess it should stay optional. Since we discussed the non-free integration on Slack, maybe @nf-core/core could also cross check.

File: nextflow_schema.json
Comment (christopher-mohr) [inline nextflow_schema.json:186]:
```suggestion
	  "description" : "External MHC binding prediction software that is not shipped with the pipeline.",
```

File: nextflow_schema.json
Comment (christopher-mohr) [inline nextflow_schema.json:192]:
```suggestion
	      "description" : "To use the 'netmhcpan' tool, specify the path to the original software tarball for NetMHCpan 4.0 (Linux) here."
```

File: nextflow_schema.json
Comment (christopher-mohr) [inline nextflow_schema.json:197]:
```suggestion
	      "description" : "To use the 'netmhc' tool, specify the path to the original software tarball for NetMHC 4.0 (Linux) here."
```

File: nextflow_schema.json
Comment (christopher-mohr) [inline nextflow_schema.json:202]:
```suggestion
	      "description" : "To use the 'netmhciipan' tool, specify the path to the original software tarball for NetMHCIIpan 3.1 (Linux) here."
```

File: nextflow_schema.json
Comment (christopher-mohr) [inline nextflow_schema.json:207]:
```suggestion
	      "description" : "To use the 'netmhcii' tool, specify the path to the original software tarball for NetMHCII 2.2 (Linux) here."
```

File: CHANGELOG.md
Comment (christopher-mohr) [inline CHANGELOG.md:10]:
```suggestion
- [#73](https://github.com/nf-core/epitopeprediction/pull/73) - Add support for the non-free netmhc tool family including netMHC, netMHCpan, netMHCII, and netMHCIIpan
```

Comment (christopher-mohr) [inline CHANGELOG.md:10]: Maybe also add the versions here

File: .github/workflows/ci.yml
Comment (christopher-mohr) [inline .github/workflows/ci.yml:94]: I would suggest to use a different name here, maybe just "Run netMHC tool tests"

**Review by christopher-mohr — CHANGES_REQUESTED** [review]
LGTM! Just some minor things from my side. Thanks! 
Let's wait for feedback from the nf-core core team regarding your question before merging.

File: conf/test_netmhcii.config
Comment (lkuchenb) [reply conf/test_netmhcii.config:23]: Yup, for some reason upstream used a different compression for this one.

File: CHANGELOG.md
Comment (lkuchenb) [reply CHANGELOG.md]: Done.

File: .github/workflows/ci.yml
Comment (lkuchenb) [reply .github/workflows/ci.yml]: Done.

**Review by christopher-mohr — APPROVED** [review]

**Pull request merged**
